### PR TITLE
chore: move testing fixtures to fixtures directory

### PR DESCRIPTION
**Pull request by cam-garrison**
move helpers to separate files

## Description

This PR follows up #899 which introduced the fixtures subdirectory to the features integration tests. 

It brings `/crds` and `/templates` into `/fixtures` and moves more helper functions out into dedicated files to store them. Goal here is to enhance readability of our test suite. 

## How Has This Been Tested?

This change only applies to our testing suite, so re-running `make` suffices to test. Ensured unit-tests still work as expected.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: tests/integration/features/features_int_test.go [hunk 1]
```diff
@@ -51,12 +49,12 @@ var _ = Describe("feature preconditions", func() {
...
 			testFeatureName := "test-ns-creation"
 			namespace = envtestutil.AppendRandomNameTo(testFeatureName)
-			dsci = newDSCInitialization(namespace)
+			dsci = fixtures.NewDSCInitialization(namespace)
 		})
 
 		It("should create namespace if it does not exist", func() {
 			// given
-			_, err := getNamespace(namespace)
+			_, err := fixtures.GetNamespace(namespace, envTestClient)
```
Comment (bartoszmajsak) [inline tests/integration/features/features_int_test.go:57]: The convention you can find in k8s golang libraries is to pass ctx,  client and then actual params. I would change the order accordingly.

File: tests/integration/features/features_int_test.go [hunk 2]
```diff
@@ -204,7 +202,7 @@ var _ = Describe("feature cleanup", func() {
 				Expect(featuresHandler.Apply()).To(Succeed())
 
 				// then
-				featureTracker, err := getFeatureTracker("always-working-feature", appNamespace)
+				featureTracker, err := fixtures.GetFeatureTracker("always-working-feature", appNamespace, envTestClient)
```
Comment (bartoszmajsak) [inline tests/integration/features/features_int_test.go:205]: Same as with `GetNamespace` - lets pass the client first.

File: tests/integration/features/features_int_test.go [hunk 3]
```diff
@@ -373,7 +371,7 @@ var _ = Describe("feature cleanup", func() {
 				Expect(featuresHandler.Apply()).To(Succeed())
 
 				// then
-				service, err := getService("knative-local-gateway", namespace.Name)
+				service, err := fixtures.GetService("knative-local-gateway", namespace.Name, envTestClient)
```
Comment (bartoszmajsak) [inline tests/integration/features/features_int_test.go:374]: I would make it `client, ns, name` here.

File: tests/integration/features/features_int_test.go [hunk 4]
```diff
@@ -482,7 +480,7 @@ metadata:
 				Expect(featuresHandler.Apply()).To(Succeed())
 
 				// then
-				cfgMap, err := getConfigMap("my-configmap", featuresHandler.ApplicationsNamespace)
+				cfgMap, err := fixtures.GetConfigMap("my-configmap", featuresHandler.ApplicationsNamespace, envTestClient)
```
Comment (bartoszmajsak) [inline tests/integration/features/features_int_test.go:483]: Again - `client, ns, name`

File: tests/integration/features/fixtures/file_sys_fixtures.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+	if err != nil {
+		return err
+	}
+
+	_, err = file.WriteString(data)
+	if err != nil {
+		return err
+	}
+	return file.Sync()
+}
```
Comment (bartoszmajsak) [inline tests/integration/features/fixtures/file_sys_fixtures.go:20]: small nit:

```suggestion
func CreateFile(dir, filename, content string) error {
	filePath := filepath.Join(dir, filename)
	file, err := os.Create(filePath)
	if err != nil {
		return err
	}

	_, err = file.WriteString(content)
	if err != nil {
		return err
	}
	return file.Sync()
}
```

**Review by bartoszmajsak — CHANGES_REQUESTED** [review]
Thanks for these improvements. There are a few nits we can take care off, but other than that it looks good!

File: tests/integration/features/features_int_test.go (hunk 4 above)
Comment (cam-garrison) [reply tests/integration/features/features_int_test.go]: 5686115

File: tests/integration/features/features_int_test.go (hunk 3 above)
Comment (cam-garrison) [reply tests/integration/features/features_int_test.go]: 5686115

File: tests/integration/features/features_int_test.go (hunk 2 above)
Comment (cam-garrison) [reply tests/integration/features/features_int_test.go]: 5686115

**Review by bartoszmajsak — APPROVED** [review]
LGTM, thanks!

**Review by zdtsw — APPROVED** [review]

**Pull request merged**